### ISSUE TITLE
Support target authentication

### ISSUE THREAD

**Issue by embano1** [knative-sandbox/kn-plugin-event#33]
The `send` command currently assumes no authentication in the endpoint. `kn event send` is useful outside of direct Knative cluster access, e.g. to send Cloud Events to arbitrary webhooks or endpoints (services) in Kubernetes/Knative which enforce authentication, e.g. basic_auth.

What do you think about adding support for authentication schemes?

**Comment by cardil** [comment]
At first, I thought this is a good idea - a common case. But, then I recalled that this issue is similar to #31. 

Seeing those two issues, I fear, a little, that after a while we might end up with having cURL number of options slipped to kn-event:

```
$ curl --help all | wc -l
241
```

I'm not sure is that a correct way to go. Maybe we could do better somehow...

Any ideas, @rhuss?

**Comment by embano1** [comment]
Yeah, the API would certainly grow. But IMHO there's a couple of basic flags which should be supported, e.g. TLS, insecure and (basic) auth. Let's see what Roland says...

**Comment by rhuss** [comment]
I'm +1 for adding security-related options as this is something that is not uncommon. At the moment I don't see that we will run into option bloat, so adding commonly used transport related options makes much sense to me.
